### ISSUE TITLE
Simpler Scenario Focused (getting started API) for ML.NET

### ISSUE THREAD

**Issue by asthana86** [dotnet/machinelearning#1800]
ML.NET is a machine learning framework focused on bringing .NET developers to the world of machine learning. Given the complexity of machine learning there are a number of concepts for developers to get familiar with before becoming successful right from the get-go.  

Is there a way to provide a more templated/recipe-driven scenario focused set of API which represent common machine learning task for developers like Recommendation, Sentiment Analysis, Forecasting, Image Classification etc. This can help them get started quickly and progressively graduate to machine learning using the Dynamic API guarenting both short-term and long term success with ML. 

e.g. (Recommendation: Just a quick ugly mockup)

Currently for the recommendation scenario for Matrix Factorization the input feature are limited to UserId, OrderId, Rating or for co-purchase scenarios limited to UserId, OrderId. Given the nature of these features the transforms (categorical) and trainers (Matrix Factorization/One Class Matrix Factorization) are somewhat pre-known. 

Is it potentially possible to simplify the API to as follows:

            //STEP 1: Create MLContext to be shared across the model creation workflow objects 
            var mlcontext = new MLContext();

            //STEP 2: Create a reader by defining the schema for reading the movie recommendation datasets
            var reader = mlcontext.Data.TextReader(new TextLoader.Arguments()
            {
                Separator = "tab",
                HasHeader = true,
                Column = new[]
                {
                    new TextLoader.Column("userId", DataKind.Text, 0),
                    new TextLoader.Column("movieId", DataKind.Text, 1),
                    new TextLoader.Column("Rating", DataKind.R4, 2)
                }
            });

            //STEP 3: Read in training data 
            var trainingDataView = reader.Read(new MultiFileSource(TrainingDataLocation));

            //STEP 4: Create default Recommendtion model 
            var model = mlcontext.CreateRecommendationModel(trainingDataView);

            //STEP 5: Read in test data 
            var testDataView = reader.Read(new MultiFileSource(TestDataLocation));

            //STEP 6: Use Recommendation model to evaluate performance of test data 
            var prediction = model.Transform(testDataView);
            var metrics = model.Evaluate(prediction, label: "Label", score: "Score");

This for getting started purposes avoids .NET developers to necessarily dive into understanding the concept of estimators, transforms, choice of learners etc. 

  var pipeline = mlcontext.Transforms.Categorical.MapValueToKey("userId", "userIdEncoded")
                                    .Append(mlcontext.Transforms.Categorical.MapValueToKey("movieId", "movieIdEncoded")
                                    .Append(new MatrixFactorizationTrainer(mlcontext, "Label", "userIdEncoded", "movieIdEncoded")));

Other ML Frameworks focused on developers and AutoML approaches out there provide a simpler set of APIs targeting developers already. Is there something we can learn on that front given ML.NET already has some AutoML smarts based upon the static rule based engine.

**Comment by xadupre** [comment]
Hi,

I explored a scenario inspired from python based on a pipeline and a C# DataFrame. It uses ml.net command line + a DataFrame based on IDataView. Here is an example with the iris dataset:

            var iris = "iris.txt";
            var df = DataFrameIO.ReadCsv(iris, sep: '\t');
            df["LabelI"] = df["Label"].AsType(NumberType.R4));
            var pipe = new ScikitPipeline(new[] { $"Concat{{col=Features:{df.Columns[1]},{df.Columns[2]}}}" }, "mlr");
            pipe.Train(df, "Features", "LabelI");
            DataFrame pred = null;
            pipe.Predict(df, ref pred);
            // var pred = pipe.Predict(df); works too

Xavier

**Comment by codemzs** [comment]
we have tons of samples here to get started and even in the samples repo

https://github.com/dotnet/machinelearning/tree/master/docs/samples/Microsoft.ML.Samples/Dynamic
